### PR TITLE
Update LineCounter to .NET 7 and use OTel instead of AppInsights

### PR DESCRIPTION
**Pull request by lmolkova**

### CODE REVIEW

File: samples/linecounter/BlobProcessorHost.cs
Comment (lmolkova) [inline samples/linecounter/BlobProcessorHost.cs:1]: renamed to `LineCounterService.cs`, no changes

**Review by rajkumar-rangaraj — APPROVED** [review]
LGTM

**Review by JoshLove-msft — APPROVED** [review]

File: samples/linecounter/Views/Shared/_Layout.cshtml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            @RenderBody()
+        </main>
+    </div>
+
+    <footer class="border-top footer text-muted">
+        <div class="container">
+            &copy; 2023 - LineCounter
+        </div>
+    </footer>
+    <script src="~/lib/jquery/dist/jquery.min.js"></script>
```
Comment (JoshLove-msft) [inline samples/linecounter/Views/Shared/_Layout.cshtml:41]: Where are we using jQuery?

File: samples/linecounter/LineCounter.csproj [hunk 2]
```diff
@@ -1,18 +1,20 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk.Web">
   <PropertyGroup>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>net7.0</TargetFramework>
```
Comment (vishweshbankwar) [inline samples/linecounter/LineCounter.csproj:3]: Could we update to `.net8.0`? It releases in couple days . That way, the additional metrics from .NET8.0 are also added.

**Review by vishweshbankwar — APPROVED** [review]

File: samples/linecounter/Views/Shared/_Layout.cshtml (hunk 1 above)
Comment (lmolkova) [reply samples/linecounter/Views/Shared/_Layout.cshtml]: good catch, removed!

File: samples/linecounter/LineCounter.csproj (hunk 2 above)
Comment (lmolkova) [reply samples/linecounter/LineCounter.csproj:3]: sure, there are a few things that need to be updated. I'd prefer to do it in a different PR, created https://github.com/Azure/azure-sdk-for-net/issues/39960

**Pull request merged**